mitiq/mitiq_pyquil/pyquil_utils.py

```diff
@@ -15,167 +15,114 @@
 from pyquil.simulation.matrices import I as npI, X as npX, Y as npY, Z as npZ
 
-QVM = get_qc("1q-qvm")
-
-# Set the random seeds for testing
-QVM.qam.random_seed = 1337
-np.random.seed(1001)
-
-
-def random_identity_circuit(depth: int) -> Program:
-    """Returns a single-qubit identity circuit based on Pauli gates."""
-
-    # initialize a quantum circuit
-    prog = Program()
+from mitiq.mitiq_pyquil.compiler import basic_compile
 
-    # index of the (inverting) final gate: 0=I, 1=X, 2=Y, 3=Z
-    k_inv = 0
 
-    # apply a random sequence of Pauli gates
-    for _ in range(depth):
-        # random index for the next gate: 1=X, 2=Y, 3=Z
-        k = np.random.choice([1, 2, 3])
-        # apply the Pauli gate "k"
-        if k == 1:
-            prog += X(0)
-        elif k == 2:
-            prog += Y(0)
-        elif k == 3:
-            prog += Z(0)
-
-        # update the inverse index according to
-        # the product rules of Pauli matrices k and k_inv
-        if k_inv == 0:
-            k_inv = k
-        elif k_inv == k:
-            k_inv = 0
-        else:
-            _ = [1, 2, 3]
-            _.remove(k_inv)
-            _.remove(k)
-            k_inv = _[0]
-
-    # apply the final inverse gate
-    if k_inv == 1:
-        prog += X(0)
-    elif k_inv == 2:
-        prog += Y(0)
-    elif k_inv == 3:
-        prog += Z(0)
-
-    return prog
-
-
-def run_with_noise(circuit: Program, noise: float, shots: int) -> float:
-    """Returns the expectation value of a circuit run several times with noise.
+def add_depolarizing_noise(pq: Program, noise: float) -> Program:
+    """Returns a quantum program with depolarizing channel noise.
 
     Args:
-        circuit: Quantum circuit as :class:`~pyquil.quil.Program`.
+        pq: Quantum program as :class:`~pyquil.quil.Program`.
         noise: Noise constant for depolarizing channel.
-        shots: Number of shots the circuit is run.
 
     Returns:
-        expval: Expectation value.
+        pq: Quantum program with added noise.
     """
+    pq = pq.copy()
     # apply depolarizing noise to all gates
     kraus_ops = [
         np.sqrt(1 - noise) * npI,
         np.sqrt(noise / 3) * npX,
         np.sqrt(noise / 3) * npY,
         np.sqrt(noise / 3) * npZ,
     ]
-    circuit.define_noisy_gate("X", [0], append_kraus_to_gate(kraus_ops, npX))
-    circuit.define_noisy_gate("Y", [0], append_kraus_to_gate(kraus_ops, npY))
-    circuit.define_noisy_gate("Z", [0], append_kraus_to_gate(kraus_ops, npZ))
-
-    # set number of shots
-    circuit.wrap_in_numshots_loop(shots)
-
-    # we want to simulate noise, so we run without compiling
-    results = QVM.run(circuit)
-    expval = (results == [0]).sum() / shots
-    return float(expval)
+    pq.define_noisy_gate("X", [0], append_kraus_to_gate(kraus_ops, npX))
+    pq.define_noisy_gate("Y", [0], append_kraus_to_gate(kraus_ops, npY))
+    pq.define_noisy_gate("Z", [0], append_kraus_to_gate(kraus_ops, npZ))
+    return pq
```

Comment:
This function adds noise only if the gate is a single qubit Pauli: X Y Z.
It was useful to scale the noise of the old RB circuits which were made only of Pauli gates.
Now RB circuits are more general and so I think we should just remove `add_depolarizing_noise`  or make it compatible with generic circuits.

Reply:
Ah yeah, I didn't realize this. To be a useful utility it should really be general, I agree. I will open a separate issue for this.